socs/agents/hwp_supervisor/agent.py

```diff
@@ -626,27 +644,88 @@ def update(self, clients, hwp_state):
             self.lock.acquire()
             state = self.action.cur_state
 
+            def query_pid_state():
+                data = self.run_and_validate(clients.pid.get_state)['data']
+                self.log.info("pid state: {data}", data=data)
+                return data
+
             if isinstance(state, ControlState.PIDToFreq):
                 self.run_and_validate(clients.pid.set_direction,
                                       kwargs={'direction': state.direction})
                 self.run_and_validate(clients.pid.declare_freq,
                                       kwargs={'freq': state.target_freq})
                 self.run_and_validate(clients.pmx.use_ext)
                 self.run_and_validate(clients.pmx.set_on)
-                self.run_and_validate(clients.pid.tune_freq)
-                #  kwargs={'freq': state.target_freq})
+                self.run_and_validate(clients.pid.tune_freq, timeout=60)
+                self.run_and_validate(
+                    clients.pcu.send_command,
+                    kwargs={'command': 'off'}, timeout=None,
+                )
+                self.action.set_state(ControlState.CheckInitialRotation(
+                    target_freq=state.target_freq,
+                    freq_tol=state.freq_tol,
+                    freq_tol_duration=state.freq_tol_duration,
+                    direction=state.direction,
+                ))
+
+            if isinstance(state, ControlState.CheckInitialRotation):
+                fhwp = hwp_state.enc_freq
+                if fhwp >= 0.2:
+                    self.action.set_state(ControlState.WaitForTargetFreq(
+                        target_freq=state.target_freq,
+                        freq_tol=state.freq_tol,
+                        freq_tol_duration=state.freq_tol_duration,
+                        direction=state.direction,
+                    ))
+                    return
+
+                if time.time() - state.start_time < state.check_wait_time:
+                    return
 
+                time.sleep(5)
```

Comment:
Why the 5 second sleep after waiting for the `check_wait_time` duration? Should `check_wait_time` just be 5 seconds longer?

Reply:
yes.... that's a good point